src/Simulation/Simulators/ToffoliSimulator/ToffoliSimulator.cs

```diff
@@ -23,7 +24,7 @@ public enum ToffoliDumpFormat
     /// <summary>
     /// The Toffoli simulator implementation class.
     /// </summary>
-    public partial class ToffoliSimulator : SimulatorBase
+    public partial class ToffoliSimulator : SimulatorBase, IQsharpCore
```

Comment:
What work is still missing for the ToffoliSimulator and the Tracer/ResourcesEstimator to process code compiled for a all three gate sets? Aside from that support, is there any other difference between how the three simulators behave/what they support?

Reply:
All that is missing is the need to implement the native gates for those packages, which include things like `ApplyUncontrolledX` and `SWAP`. This isn't hard, I just didn't think it was necessary for the current PR. I agree that we should consider supporting all targets on these simulators, but there are some open questions, namely:

- How do we want to refactor the ResourcesEstimator, and will that change the approach and how we support multiple targets?

- Does it make sense to represent ToffoliSimulator as it's own target package, since it defines a gate set, rather than allow someone to use a specific gate set and then target Toffoli which doesn't actually support that set?